Port TH library to ATen/Parallel

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #20157 Support for Eigen thread pool
* #20087 Native ATen/Parallel backend
* #20057 Split ATen/Parallel into interface and backend
* #20050 Move inter-op settings into ATen/Parallel
* #20043 Port ATen/native to ATen/Parallel
* #20032 Port THNN to ATen/Parallel
* #20002 Remove explicit checks for parallelism from TH
* **#19105 Port TH library to ATen/Parallel**

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

Differential Revision: [D14947557](https://our.internmc.facebook.com/intern/diff/D14947557)

Commit:
Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Move OMP/MKL thread initialization into ATen/Parallel"

Move OMP/MKL thread initialization into ATen/Parallel

Summary:
This PR implements changes described in
https://github.com/pytorch/pytorch/issues/19001

gh-metadata: pytorch pytorch 19011 gh/ilia-cher/1/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update base for Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Initialize intra-op threads in JIT thread pool"

Initialize intra-op threads in JIT thread pool

Summary:
Similar to initialization in torch initModule and autograd's
Engine::thread_init, we need to call at::init_num_threads when
initializing JIT thread pool threads

gh-metadata: pytorch pytorch 19058 gh/ilia-cher/2/head

Commit:
Update base for Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

ilia-cher commented:
fixed a bug with accumulator type in TH_TENSOR_APPLY_REDUCTION_SUM_PARALLEL

cpuhrsch commented on aten/src/TH/generic/THTensorEvenMoreMath.cpp:
```diff
@@ -71,12 +71,12 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
   int mask_contig = THTensor_(isContiguous)(mask);
-  if (!omp_in_parallel() && tensor_contig && mask_contig) {
-    TH_TENSOR_APPLY2_OMP(tensor_size, tensor_contig, mask_contig,
+  if (!at::in_parallel_region() && tensor_contig && mask_contig) {
```
Isn't this already handled by at::parallel_for itself? Or should we add this for future compatibility? I'd say it's up to the threadpool library whether or not it provides good support for nested parallelism.

ilia-cher replied:
right, we seem to be doing this check twice, will update

cpuhrsch commented on aten/src/TH/generic/THTensorApply.hpp:
```diff
@@ -1,41 +1,323 @@
+  if (CONTIG1 && CONTIG2) { \
+    TYPE1 *rp = THTensor_getStoragePtr(TENSOR1)->data<TYPE1>()+TENSOR1->storage_offset(); \
+    TYPE2 *tp = THTensor_getStoragePtr(TENSOR2)->data<TYPE2>()+TENSOR2->storage_offset(); \
+    if (tp != (TYPE2*)rp) { \
```
I wonder if we can go a step further for these and start using the TensorIterator. We should probably do that in a follow-up PR however, unless it can significantly simplify this code.

cpuhrsch replied:
cc @VitalyFedyunin

cpuhrsch replied:
One of the risks however is that we're starting to introduce too many dependencies on ATen. One of the opportunities is to move a lot of these ops into an e.g. legacy_implementation.cpp within aten/native and delete them from TH all-together. They can then be refactored into something nicer down the road.

ilia-cher replied:
tbh my goal in this PR was purely porting to ATen/Parallel primitive without much changes to TH, to speed up transition away from OpenMP and minimize chance of introducing new bugs. We can think about using TensorIterator in the next PRs, or deprecate TH altogether

cpuhrsch commented on aten/src/TH/generic/THTensorApply.hpp:
```diff
@@ -1,41 +1,323 @@
+  } \
+}
+
+#define TH_TENSOR_APPLY3_PARALLEL(SIZE, CONTIG1, CONTIG2, CONTIG3, TYPE1, TENSOR1, TYPE2, TENSOR2, TYPE3, TENSOR3, CODE, THRESHOLD) \
```
I assume this code is a copy-paste from the original macros with openmp replaced with at::parallel_for

ilia-cher replied:
correct, with some minimum modifications caused by introduction of lambdas

cpuhrsch commented on aten/src/ATen/Parallel.h:
```diff
@@ -7,6 +7,8 @@
 #include <exception>
 
 #ifdef _OPENMP
+#define INTRA_OP_PARALLEL
```
Does it make sense to consider moving this into parallel_for, or having parallel_for switch to a serial implementation? This could also help remove a lot of TH code. I wonder if we currently have users who actively disable OpenMP and actually use these non-parallelized branches. Of course this is separate from simply setting the number of threads to 1.

In fact, maybe, being able to branch on the number of threads to a serial implementation within at::parallel could facilitate better code generation, because the compiler only needs to deal with a single for-loop instead and can avoid the OpenMP branch all-together.

ilia-cher replied:
My take was that if user wants, for some reason, to remove parallel code from compilation completely, then I should support this, so I just simply replaced _OPENMP with INTRA_OP_PARALLEL. But I see your point, maybe we don't even need this too, I'll take another look at the use cases of _OPENMP, I think some of them could be removed and hidden inside parallel_for yes

cpuhrsch reviewed (changes requested):
Requesting answer to comments

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

ilia-cher commented:
rebased on top of  https://github.com/pytorch/pytorch/pull/19997

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

ilia-cher commented:
addressed @cpuhrsch comments as an extra step in https://github.com/pytorch/pytorch/pull/20002 - I checked the speed with https://github.com/pytorch/pytorch/pull/19997 and it seems we should be good

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

VitalyFedyunin reviewed (approved):
--- nvm ----

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head

Commit:
Update on "Port TH library to ATen/Parallel"

Port TH library to ATen/Parallel

Summary:
Removing explicit usage of OpenMP from TH.
More details in https://github.com/pytorch/pytorch/issues/19002

Test Plan:
BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
pytest -s -v test/test_torch.py::TestTorch

gh-metadata: pytorch pytorch 19105 gh/ilia-cher/3/head
